src/cli/Peers.ts

```diff
@@ -0,0 +1,387 @@
+        }
+      }),
+    );
+}
+function makeGetPeerInfoCommand() {
+  return new commander.Command('get')
+    .description('get the peer info for a particular public key')
+    .option('--node-path <nodePath>', 'node path')
+    .option('-b, --base64', 'output peer info as a base64 string')
+    .option('-c, --current-node', 'only list the peer information for the current node, useful for sharing')
+    .option('-p, --public-key <publicKey>', 'path to the file which contains the public key')
+    .option('-v, --verbose', 'increase verbosity level by one')
+    .action(
+      actionRunner(async (options) => {
+        const client = PolykeyAgent.connectToAgent();
+        const status = await client.getAgentStatus();
+        if (status != 'online') {
+          throw Error(`agent status is: ${status}`);
+        }
+
+        const nodePath = determineNodePath(options);
+        const currentNode: boolean = options.currentNode;
+        const publicKeyPath = options.publicKey;
+        const base64 = options.base64;
+        // read in publicKey if it exists
+        let publicKey: string | undefined;
+        if (publicKeyPath) {
+          publicKey = fs.readFileSync(publicKeyPath).toString();
+        }
+
+        const peerInfo: PeerInfo = await client.getPeerInfo(nodePath, currentNode, publicKey);
+
+        if (base64) {
+          pkLogger(peerInfo.toStringB64(), PKMessageType.SUCCESS);
+        } else {
+          pkLogger('Peer Public Key:', PKMessageType.INFO);
+          pkLogger(peerInfo.publicKey, PKMessageType.SUCCESS);
+
+          pkLogger('Peer Connected Address:', PKMessageType.INFO);
+          pkLogger(peerInfo.connectedAddress?.toString() ?? '', PKMessageType.SUCCESS);
+
+          pkLogger('Relay Public Key:', PKMessageType.INFO);
+          pkLogger(peerInfo.relayPublicKey ?? '', PKMessageType.SUCCESS);
+        }
+      }),
+    );
+}
+
+function makeListPeersCommand() {
+  return new commander.Command('list')
+    .description('list all connected peers')
+    .alias('ls')
+    .option('--node-path <nodePath>', 'node path')
+    .option('-v, --verbose', 'increase verbosity level by one')
+    .action(
+      actionRunner(async (options) => {
+        const client = PolykeyAgent.connectToAgent();
+        const status = await client.getAgentStatus();
+        if (status != 'online') {
+          throw Error(`agent status is: ${status}`);
+        }
+
+        const nodePath = determineNodePath(options);
+
+        const publicKeys = await client.listPeers(nodePath);
+        if (publicKeys === undefined || publicKeys.length == 0) {
+          pkLogger('no peers exist', PKMessageType.INFO);
+        } else {
+          publicKeys.forEach((publicKey: string, index: number) => {
+            pkLogger(`${index + 1}: ${publicKey}`, PKMessageType.INFO);
+          });
+        }
+      }),
+    );
+}
+
+function makePingPeerCommand() {
+  return new commander.Command('ping')
+    .description('ping a connected peer')
```

Comment:
Is this meant to map to our UI actions like `scan` and `listen`... etc. Or is this just a debug command? It should be clearer, and debug commands should always be lower prioritized in the order of commands being added and semantic grouping in the code. Remember that we are using sonar analogy for p2p, but things like scan for vault sharing. Remind me if I have used these terms wrong. We need this for our UI flow diagrams.

Reply:
yeah this is not related to our `sonar ping` which is about finding all peers currently listening on the network. So this should be changed.